src/dune_lang/predicate_with_id.mli

```diff
@@ -2,10 +2,15 @@
     string. The uniquely identifying string allows us to safely memoize on the
     predicate *)
 
+(* TODO move back to dune_engine? this is really only needed for
+   [File_selector.t] *)
```

Comment:
(Just noticed this TODO while importing recent changes internally.) Oh yes, we should definitely inline `Predicate_with_id` into `File_selector`.